aws/resource_aws_lambda_function.go

```diff
@@ -25,29 +25,11 @@ import (
 
 const awsMutexLambdaKey = `aws_lambda_function`
 
-var validLambdaRuntimes = []string{
+var validateLambdaRuntime = validation.All(
+	validation.StringInSlice(lambda.Runtime_Values(), false),
 	// lambda.RuntimeNodejs has reached end of life since October 2016 so not included here
-	lambda.RuntimeDotnetcore10,
-	lambda.RuntimeDotnetcore20,
-	lambda.RuntimeDotnetcore21,
-	lambda.RuntimeDotnetcore31,
-	lambda.RuntimeGo1X,
-	lambda.RuntimeJava8,
-	lambda.RuntimeJava11,
-	lambda.RuntimeNodejs43,
-	lambda.RuntimeNodejs43Edge,
-	lambda.RuntimeNodejs610,
-	lambda.RuntimeNodejs810,
-	lambda.RuntimeNodejs10X,
-	lambda.RuntimeNodejs12X,
-	lambda.RuntimeProvided,
-	lambda.RuntimePython27,
-	lambda.RuntimePython36,
-	lambda.RuntimePython37,
-	lambda.RuntimePython38,
-	lambda.RuntimeRuby25,
-	lambda.RuntimeRuby27,
-}
+	validation.StringNotInSlice([]string{lambda.RuntimeNodejs}, false),
```

Comment:
Retained for backwards compatibility but given that there are now many more runtimes that are at end-of-life we may want to remove this additional check.

Reply:
Let's go ahead and remove the "all but" testing of these -- if AWS wants to officially remove support for these, they can properly remove the values from the enumeration type in the API model 👍

Reply:
The provider documentation doesn't explicitly list the values for this attribute but instead references https://docs.aws.amazon.com/lambda/latest/dg/API_PublishLayerVersion.html#SSS-PublishLayerVersion-request-CompatibleRuntimes which shows all the values in the API enumeration, so there's nothing to change there.
I will ensure that the examples in the documentation use currently valid values (I saw a reference to `nodejs8.10`).

Reply:
Done.